crates/polars-core/src/chunked_array/ops/lambda.rs

```diff
@@ -15,6 +15,7 @@ pub enum LambdaExpression {
     GreaterThan(Box<Self>, Box<Self>),
     LessThan(Box<Self>, Box<Self>),
     IfThenElse(Box<Self>, Box<Self>, Box<Self>),
+    CaseWhen(Vec<(Box<Self>, Box<Self>)>, Box<Self>)
```

Comment:
I have only one comment, I think the tuple doesn't need to be Boxed, Vec is already heap-allocated, so the enum will allow it
(the otherwise should remain Boxed)

Reply:
(This means another repeated dereferencing as it runs for each element, for some cases the compiler will optimize this, for other cases this can be a slight performance hit which we don't need